crates/ibc/src/core/ics02_client/events.rs

```diff
@@ -390,3 +390,101 @@ impl From<UpgradeClient> for abci::Event {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::mock::client_state::{client_type as mock_client_type, MOCK_CLIENT_TYPE};
+    use crate::mock::header::MockHeader;
+    use ibc_proto::google::protobuf::Any;
+    use tendermint::abci::Event as AbciEvent;
+
+    #[test]
+    fn ibc_to_abci_client_events() {
+        struct Test {
+            kind: IbcEventType,
+            event: AbciEvent,
+            expected_keys: Vec<&'static str>,
+            expected_values: Vec<String>,
+        }
+
+        let client_type = mock_client_type();
+        let client_id = ClientId::new(client_type.clone(), 0).unwrap();
+        let consensus_height = Height::new(0, 10).unwrap();
+        let consensus_heights = vec![Height::new(0, 5).unwrap(), Height::new(0, 7).unwrap()];
+        let header: Any = MockHeader::new(consensus_height).into();
+        let expected_keys = vec![
+            CLIENT_ID_ATTRIBUTE_KEY,
```

Comment:
Can we hardcode the value `"client_id"`, and a [URL to a the ibc-go value](https://github.com/cosmos/ibc-go/blob/887cba1658bbb2db46b96d1ae492cbe8095b1871/modules/core/03-connection/types/events.go#L10-L15) above the definition of `expected_keys`? This is to avoid the case that we change the value of `CLIENT_ID_ATTRIBUTE_KEY` for some reason; the test wouldn't catch that.

The general principle in these sorts of tests is you want to reuse as little as possible of the definitions that the implementation uses; otherwise you run the risk of changing the implementation incorrectly, and changing the test in the same way such that it doesn't fail when it should have.

Reply:
Well said 👍🏻 
Fixed https://github.com/cosmos/ibc-rs/pull/363/commits/ffa0bbcf972b93e074829390cb9cd3511eae23f9